.github/linters/.hadolint.yml

```diff
@@ -1,2 +1,4 @@
+failure-threshold: warning
 ignored:
   - DL3041
+  - DL3013
```

Comment:
Yea, I got playing around on this, and ended up kinda getting a bit too crazy here already I think. It's been like this since Miguel changed it for downstream purposes, and it seems like hadolint is just now starting to find this? I re-ran the `pip-compile` command which complains about the unsafe setup we have, and then I also saw that we have an upgrade step in the `Dockerfile` which doesn't actually do anything, because the `requirements-build.txt` just ends up reverting the upgrade anyways (to hit the target release).

I'm thinking I might take your suggestion, but I might just break all this work out into a separate lint fix setup. I don't think we're dealing with this correctly, and I'm not really sure why we're doing it this way since it seems to be unsafe. I guess my fear is that I change things here and then break things during import for some reason.

Reply:
I'm going to track and address this as part of https://issues.redhat.com/browse/STF-1256 because I think there is enough clean up here to justify a separate tracking issue.